### PR TITLE
Add use cases to DG

### LINKED ISSUE #5: Add use cases to DG
- [x] Use cases: Give use cases (textual form) for a few representative user stories that need multiple steps to complete. e.g. Adding a tag to a person (assume the user needs to find the person first)

### PR DESCRIPTION
**Pull request by yanchenglee98**
Add 5 use cases to DG

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -262,32 +262,88 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
-(For all use cases below, the **System** is the `AddressBook` and the **Actor** is the `user`, unless specified otherwise)
+(For all use cases below, the **System** is the `AddressBook` and the **Actor** is the `hall leader`, unless specified otherwise)
 
-**Use case: Delete a person**
+**Use case: UC01 - Listing of contacts**
 
 **MSS**
+1.  User requests to list residents
+2.  AddressBook shows a list of residents
+Use case ends
```
Comment (schoolex) [inline docs/DeveloperGuide.md:273]: Are you missing a new line here?

**Review by schoolex — COMMENTED** [review]
Minor bug in formatting

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -262,32 +262,89 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+1.  User enters resident's details
+2.  AddressBook adds resident into list
+3.  AddressBook <ins>displays list (UC01)</ins>
+
+    Use case ends.
+
+**Extensions**
+
+* 1a. User eneters details in the wrong format.
+    * 1a1. AddressBook shows an error message and correct format.
```
Comment (teekoksiang) [inline docs/DeveloperGuide.md:310]: Should there be a new line between 1a and 1a1? This is also applicable to line 334 and 335.

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -262,32 +262,89 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+**Extensions**
+
+* 1a. The list is empty.
+
+  Use case ends.
+
+* 2a. The given index is invalid.
+
+    * 2a1. AddressBook shows an error message.
```
Comment (teekoksiang) [inline docs/DeveloperGuide.md:333]: Should there be an indication of the next step for the use case?

**Review by teekoksiang — COMMENTED** [review]
Minor bug in formatting

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (yanchenglee98) [reply docs/DeveloperGuide.md:310]: I think no need since a new line is already shown between 1a and 1a1 when viewed in markdown.

**Review by teekoksiang — APPROVED** [review]
LGTM

**Pull request merged**